relay-server/src/actors/events.rs

```diff
@@ -1113,14 +1113,17 @@ impl EventProcessor {
     }
 
     /// Run dynamic sampling rules to see if we keep the event or remove it.
-    fn sample_event(&self, state: &mut ProcessEnvelopeState) -> Result<(), ProcessingError> {
+    fn sample_event(
+        &self,
+        state: &mut ProcessEnvelopeState,
+        processing_enabled: bool,
```

Comment:
```suggestion
```

There should be no need to pass this flag, since you have access to `self.config.processing_enabled()` within this function.